### PR TITLE
Change Su-33 carrier flights to "Takeoff from runway"

### LINKED ISSUE #1352: AI Su-33 Fails to take off from Kuznetsov
Before filing, please search the issue tracker to see if the issue has already been reported.

If reporting a DCS AI bug, check https://github.com/dcs-liberation/dcs_liberation#dcs-bugs.

**Describe the bug**
AI controlled Su-33s are unable to take off from Admiral Kuznetsov. They are set to "take off from ramp" and are spawned partially clipping into the raised blast shields.

**To Reproduce**
Create any campaign with Su-33s and Admiral Kuznetsov (without supercarrier). Assign Su-33's to a mission. 

**Expected behavior**
Su-33's spawn without clipping and are able to take off.

**Additional information**

While this is probably a DCS bug with the free Admiral Kuznetsov there is a workaround that would allow Su-33s to use the free Admiral Kuznetsov module. If the 
take off from ramp
take off from runway
[Missions.zip](https://github.com/dcs-liberation/dcs_liberation/files/6716028/Missions.zip)
Su-33s take off waypoint is set to "take off from runway" instead of "take off from ramp", that Liberation currently uses by default, the Su-33s spawn without clipping and are able to take off.

- DCS Liberation save file (the `.liberation` file you save from the DCS Liberation window). By default these are located in your DCS saved games directory (`%USERPROFILE%/Saved Games/DCS`).
- The generated mission file (the `.miz` file that you load in DCS to play the turn). By default these are located in your missions directory (`%USERPROFILE%/Saved Games/DCS/Missions`).
- A tacview track file, especially when demonstrating an issue with AI behavior. By default these are located in your Tacview tracks directory (`%USERPROFILE%/Documents/Tacview`).
- The state.json file from the finished mission when the problem is related to results processing. By default these are located in your Liberation install directory.

**Version information (please complete the following information):**
 - DCS Liberation [e.g. 2.3.1]:

**Additional context**

### PR DESCRIPTION
**Pull request by MetalStormGhost**
Change the start_type for generated Su-33 flights from carriers to "Takeoff from runway" to work around a DCS AI issue preventing Su-33s from taking off when set to "Takeoff from ramp".

I ran into the issue reported earlier (#1352) in a coop campaign with a friend last night and I can confirm the problem. I made a really simple change to try and work around it. I did test that it generates the correct flights by opening the generated mission in the mission editor, but didn't test it by actually running the mission yet (will do that today).

Also, I wanted to create a PR for this to get advice how to properly access game.settings (or the equivalent from the settings rework that might be happening in the background) from within packagebuilder. I was thinking of making this an option, since I'm hopeful the actual DCS issue might get fixed at some point and we wouldn't need this after that anymore.

### CODE REVIEW

**Comment by DanAlbert** [comment]
> I was thinking of making this an option, since I'm hopeful the actual DCS issue might get fixed at some point and we wouldn't need this after that anymore.

When that day comes we can just remove the workaround. I don't think an option is needed here, since the option is effectively "do you want broken AI Su-33s or not?". No one does, so no option needed :)

**Review by DanAlbert — APPROVED** [review]
I see this is still a draft, but LGTM. Will avoid merging before you undraft it though.

**Comment by MetalStormGhost** [comment]
Yeah, I read through the issue one more time and realized this problem probably only applies to the free Kuznetsov. I'll change it so the supercarrier start types are not affected.

**Comment by MetalStormGhost** [comment]
I think I need some advice from someone with a bit more insight on the inner workings of Liberation. I tried checking if the carrier is a free Kuznetsov (as opposed to the supercarrier one), but I just realized that upgrade_to_supercarrier is run after packagebuilder does its thing.

Should the whole thing be moved away from packagebuilder to a stage that is run upon mission generation?

**Comment by MetalStormGhost** [comment]
I moved the logic from packagebuilder to aircraft.py and it seems to work based on my initial testing. I'm removing the draft status.

File: gen/aircraft.py [hunk 1]
```diff
@@ -265,6 +266,25 @@ def _start_type(start_type: str) -> StartType:
...
+            unit_type.id == Su_33.id
+            and group_units[1] is not None
+            and group_units[1]["type"] == KUZNECOW.id
+        ):
+            return StartType.Runway
+        elif start_type == "Runway":
+            return StartType.Runway
+        elif start_type == "Cold":
+            return StartType.Cold
+        return StartType.Warm
```
Comment (DanAlbert) [inline gen/aircraft.py]: The else branch can just be delegated to `_start_type`

File: gen/aircraft.py [hunk 2]
```diff
@@ -265,6 +266,25 @@ def _start_type(start_type: str) -> StartType:
             return StartType.Cold
         return StartType.Warm
 
+    @staticmethod
+    def _start_type_at_group(
+        start_type: str,
+        unit_type: Type[FlyingType],
+        at: Union[ShipGroup, StaticGroup],
+    ) -> StartType:
+        group_units = at.dict().get("units")
```
Comment (DanAlbert) [inline gen/aircraft.py]: This should be `at.units`, guarded with an `isinstance` if needed. We shouldn't be dependent on the serialization format of the miz.

**Review by DanAlbert — CHANGES_REQUESTED** [review]

**Comment by MetalStormGhost** [comment]
Replaced at.dict().get("units") with at.units as requested. I did not add an isinstance check as I couldn't see where that would be needed in this case.

Also removed the redundant code, now calling _start_type instead.

**Review by DanAlbert — APPROVED** [review]

**Comment by DanAlbert** [comment]
> I did not add an isinstance check as I couldn't see where that would be needed in this case.

I wasn't sure if it would be needed by mpyy (if `StaticGroup` didn't also have a `units` member). Looks like no, so not needed.

**Pull request merged**